roles/test-utils/mining-device/src/main.rs

```diff
@@ -595,17 +595,21 @@ impl Miner {
         self.header = Some(header);
     }
     pub fn next_share(&mut self) -> NextShareOutcome {
-        let header = self.header.as_ref().unwrap();
-        let mut hash = header.block_hash().as_hash().into_inner();
-        hash.reverse();
-        let hash = Uint256::from_be_bytes(hash);
-        if hash < *self.target.as_ref().unwrap() {
-            info!(
-                "Found share with nonce: {}, for target: {:?}, with hash: {:?}",
-                header.nonce, self.target, hash,
-            );
-            NextShareOutcome::ValidShare
+        if let Some(header) = self.header.as_ref() {
+            let mut hash = header.block_hash().as_hash().into_inner();
+            hash.reverse();
+            let hash = Uint256::from_be_bytes(hash);
+            if hash < *self.target.as_ref().unwrap() {
+                info!(
+                    "Found share with nonce: {}, for target: {:?}, with hash: {:?}",
+                    header.nonce, self.target, hash,
+                );
+                NextShareOutcome::ValidShare
+            } else {
+                NextShareOutcome::InvalidShare
+            }
         } else {
+            std::thread::yield_now();
```

Comment:
what is the purpose of this `yield_now`?

Reply:
avoid entering a spin loop, this i better than the hint as we can just wait here as it can happen only when the application is starting, the there will always be an header

Reply:
what do you mean by hint?

Reply:
https://doc.rust-lang.org/std/hint/index.html in that case a spin_loop hint